sdk/cognitivelanguage/Azure.AI.Language.Conversations/samples/Sample5_AnalyzeConversation_ConversationPii.md

```diff
@@ -0,0 +1,163 @@
+
+AnalyzeConversationOperationState operationResults = analyzeConversationOperation.Value;
+
+foreach (ConversationPiiOperationResult task in operationResults.Actions.Items.Cast<ConversationPiiOperationResult>())
+{
+    Console.WriteLine($"Operation name: {task.Name}");
+
+    foreach (ConversationalPiiResultWithResultBase conversation in task.Results.Conversations)
+    {
+        Console.WriteLine($"Conversation: #{conversation.Id}");
+        Console.WriteLine("Detected Entities:");
+        foreach (ConversationPiiItemResult item in conversation.ConversationItems)
+        {
+            foreach (NamedEntity entity in item.Entities)
+            {
+                Console.WriteLine($"Category: {entity.Category}");
+                Console.WriteLine($"Subcategory: {entity.Subcategory}");
+                Console.WriteLine($"Text: {entity.Text}");
+                Console.WriteLine($"Offset: {entity.Offset}");
+                Console.WriteLine($"Length: {entity.Length}");
+                Console.WriteLine($"Confidence score: {entity.ConfidenceScore}");
+                Console.WriteLine();
+            }
+        }
+        if (conversation.Warnings != null && conversation.Warnings.Any())
+        {
+            Console.WriteLine("Warnings:");
+            foreach (dynamic warning in conversation.Warnings)
+            {
+                Console.WriteLine($"Code: {warning.Code}");
+                Console.WriteLine($"Message: {warning.Message}");
+            }
+        }
+        Console.WriteLine();
+    }
+    if (operationResults.Errors != null && operationResults.Errors.Any())
+    {
+        Console.WriteLine("Errors:");
+        foreach (dynamic error in operationResults.Errors)
+        {
+            Console.WriteLine($"Error: {error}");
+        }
+    }
+}
+```
+
+## Asynchronous
+
+Using the same `data` definition above, you can make an asynchronous request by calling `AnalyzeConversationAsync`:
+
+```C# Snippet:AnalyzeConversationAsync_ConversationPii
+var data = new AnalyzeConversationOperationInput(
+    new MultiLanguageConversationInput(
+        new List<ConversationInput>
+        {
+            new TextConversation("1", "en", new List<TextConversationItem>()
+            {
+                new TextConversationItem("1", "Agent_1", "Can you provide you name?"),
+                new TextConversationItem("2", "Customer_1", "Hi, my name is John Doe."),
+                new TextConversationItem("3", "Agent_1", "Thank you John, that has been updated in our system.")
+            })
+        }),
+        new List<AnalyzeConversationOperationAction>
+        {
+            new PiiOperationAction()
+            {
+                ActionContent = new ConversationPiiActionContent(),
+                Name = "Conversation PII task",
+            }
+        });
+
+Response<AnalyzeConversationOperationState> analyzeConversationOperation = await client.AnalyzeConversationOperationAsync(data);
```

Comment:
Do we have options to change timeouts for async jobs? Or details on when they timeout? Any details around client side retries etc

Reply:
I'm also not sure if hiding all the polling behind an API is a good idea, is this how the pattern in our SDK has always existed? Do customers have an option of manually polling if they want to

Reply:
The auto-generated client already provides the APIs which allow submission/get status/cancellations individually. Check the AnalyzeConversationsClient in the generated folder. This particuloar convenience method is required for all C# SDK for customers who do not want to handle the operations for LRO. This is also generally auto-generated, but due to some issues with our TypeSpec generation, its not getting auto generated so we have been advised to hand write this.